modules/src/core/ics26_routing/handler.rs

```diff
@@ -46,13 +45,16 @@ pub fn decode(message: Any) -> Result<Ics26Envelope, Error> {
 /// Top-level ICS dispatch function. Routes incoming IBC messages to their corresponding module.
 /// Returns a handler output with empty result of type `HandlerOutput<()>` which contains the log
 /// and events produced after processing the input `msg`.
+/// If this method returns an error, the runtime is expected to rollback all state modifications to
+/// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
 pub fn dispatch<Ctx>(ctx: &mut Ctx, msg: Ics26Envelope) -> Result<HandlerOutput<()>, Error>
```

Comment:
We could make `dispatch()` private now that `deliver()` deals with one message at a time, but I would suggest we wait for #1886 (because `dispatch()` currently returns `HandlerOutput` only on error and that might change). `decode()` needs to stay public until we have a `CheckTx` equivalent (will open an issue for that as well).